### ISSUE TITLE
not-found edge case? example needed

### ISSUE THREAD

**Issue by Zerebokep** [i18nexus/next-i18n-router#32]
Could you please add an not-found.tsx locale sample to https://github.com/i18nexus/next-i18n-router/tree/main/examples/i18next-example ?

I'm not sure if I'm doing something wrong, but I always receive the following error when trying to access the locale parameter:

```
 TypeError: Cannot read properties of undefined (reading 'locale')
    at NotFound (not-found.tsx:21:37)
    at stringify (<anonymous>)
  15 | 
  16 | async function NotFound({
> 17 |   params: { locale },
     |            ^
  18 | }: {
  19 |   params: { locale: string }
  20 | }) {
```

**Comment by i18nexus** [comment]
Thanks for the feedback, we’re taking a look at this and will provide an example soon.

**Comment by i18nexus** [comment]
Unfortunately there are 2 issues that will cause problems here:

1. not-found.js does not have any props. So you cannot access the current locale via props. You can read more here: https://github.com/vercel/next.js/discussions/43179 But in this case you can use the 'x-next-i18n-router-locale` header to read the current locale. 
2. But the more major issue is that not-found.js does not seem to work inside of a dynamic segment. We have tested in a brand new project that does not use next-i18n-router and `not-found` is not rendered when nested in a dynamic segment.

In the meantime, you may want to use a catch-all page to catch your not-found pages. You can see how the catch-all strategy works here: https://stackoverflow.com/a/75625136

Since these are issues with Next.js and not this library, we are going to close this for now. We will likely open an issue with Next.js to fix #2 above.

**Comment by i18nexus** [comment]
For a full example for a workaround to implement and localize not-found pages, please see #36
